src/main/java/cubicchunks/asm/mixin/selectable/client/MixinRenderChunk_NoOptifine.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.asm.mixin.selectable.client;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.Constant;
+import org.spongepowered.asm.mixin.injection.ModifyConstant;
+
+import cubicchunks.client.IRenderChunk;
+import cubicchunks.client.RenderVariables;
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.client.renderer.chunk.RenderChunk;
+import net.minecraftforge.common.ForgeModContainer;
+/**
+ * Fixes renderEntities crashing when rendering cubes
+ * that are not at existing array index in chunk.getEntityLists(),
+ * <p>
+ * Allows to render cubes outside of 0..256 height range.
+ */
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+@Mixin(RenderChunk.class)
+public abstract class MixinRenderChunk_NoOptifine implements IRenderChunk {
```

Comment:
Could you put both `MixinRenderChunk_NoOptifine` and `MixinRenderChunk_OptifineSpecific` as inner classes in one class?

Reply:
I probably could. Will it work with config? I will need to use `client.MixinRenderChunk$NoOptifine` in mixin config and in mixins Jsons? Do I need to repeat `@Mixin(RenderChunk.class)` before every class declaration?

Reply:
`MixinRenderChunk` is in `core` package. It does not use mixin config. Moving it to `selectable` will unnecessary complexify stuff, IMHO.

Reply:
To mixin inner classes are just normal classes. So I think the answer to both is "yes".